src/states/node.rs

```diff
@@ -2375,7 +2388,14 @@ impl Node {
         info!("{:?} Other section merge completed. Prefixes: {:?}",
               self,
               self.routing_table().prefixes());
-        self.merge_if_necessary(outbox);
+
+        // This is an optimisation to avoid un-necessary merge in the scenario:
+        //      we only know one of the subsections that got merged, and that subsection contains
+        //      less than min_section_size nodes.
```

Comment:
This should probably mention that that section doesn't _really_ contain too few nodes: we just didn't learn about all its members because the message was empty?